query-advisor/from-insight-to-conclusion.mdx

```diff
@@ -0,0 +1,199 @@
+
+The insight appeared on the Query Advisor landing page with sample runtime data and affected query details.
+
+![Query Advisor landing page showing detected optimization opportunities with runtime impact analysis](query-advisor-landing.png)
+
+## Step 2: Create workbook for testing
+
+From the Query Advisor insight, we clicked **"Create Workbook"** to begin systematic testing:
+
+### Workbook setup
+1. **Query import**: Automatically populated with the problematic query
+2. **Parameter detection**: Identified `database_id` parameter from query samples
+3. **Sample data**: Selected representative parameter values from historical executions
+
+### Parameter set selection
+We chose a representative test case to validate the optimization based on the actual query parameters:
+- **Parameter Set 1**: `server_id` and `hidden` values from production workload
+- **Historical performance**: Query showing 495.51ms baseline runtime
+
+![Workbooks parameter set collection](workbooks-parameter-set-selection.png)
+
+Testing with real parameter values ensures the optimization reflects actual production conditions.
+
+## Step 3: Record baseline performance
+
+Before applying any changes, we recorded baseline performance using the **collector workflow**:
+
+### Workbooks testing workflow
+Query Advisor integrates with pganalyze Workbooks to provide systematic optimization testing:
+
+### Baseline results
+- **Runtime**: 495.51ms execution time
+- **Total Est. Cost**: 39,259 cost units
+- **I/O Read Time**: 0.00ms (warm cache scenario)
+
+The baseline confirmed the issue was reproducible and not just a one-time occurrence due to cold caches or concurrent load.
+
+![Workbooks baseline plan showing initial query performance before optimization](workbooks-baseline-plan.png)
+
+### Complete testing workflow
+The systematic approach to testing Query Advisor optimizations includes:
+
+1. **Baseline recording** - Capture current performance as reference point
+2. **Apply insights** - Use suggested rewrites to create variants
+3. **Benchmark results** - Compare performance improvements systematically
+4. **Validate changes** - Ensure optimizations work across parameter sets
+5. **Apply to production** - Implement verified improvements with confidence
+
+## Step 4: Apply Query Advisor insight
+
+In the workbook variant editor, Query Advisor provided an **"Apply"** button for the detected insight:
+
+![Workbooks interface showing Query Advisor sidebar with apply insight functionality](workbooks-query-advisor-sidebar.png)
+
+![Workbooks variant editor for applying Query Advisor insights to create optimized query versions](workbooks-query-advisor-apply-insight.png)
+
+### Automatic rewrite
+Query Advisor automatically modified the query to force better index selection:
+
+<CodeBlock language="sql">
+{`SELECT databases.*
+FROM databases
+WHERE databases.server_id = $1
+  AND databases.hidden = $2
+ORDER BY databases.id + $zero ASC  -- Added +0 to prevent index sort usage
```

Comment:
I think it's a bit confusing to see the normal param refs ($1, $2) and pganalyze named param refs ($zero). Might be good to use what we can see in the screenshot (pganalyze param refs, like $database_id) for the consistency?